sv.c

```diff
@@ -3308,7 +3308,22 @@ Perl_sv_2pv_flags(pTHX_ SV *const sv, STRLEN *const lp, const U32 flags)
         Move(ptr, s, len, char);
         s += len;
         *s = '\0';
-        SvPOK_on(sv);
+        /* We used to call SvPOK_on(). Whilst this is fine for (most) Perl code,
+           it means that after this stringification is cached, there is no way
+           to distinguish between values originally assigned as $a = 42; and
+           $a = "42"; (or results of string operators vs numeric operators)
+           where the value has subsequently been used in the other "context"
```

Comment:
I'd try to avoid the word "context" here. Maybe "subsequently been used in the other sense"

Reply:
Agree both with avoiding *context*, and I can't see a better word than *sense* so went with this.